sbncode/CAFMaker/FillReco.cxx

```diff
@@ -138,11 +138,11 @@ namespace caf
   }
 
   void FillCRTPMTMatch(const sbn::crt::CRTPMTMatching &match,
+		       const sbn::crt::MatchedCRT &crtmatch,
```

Comment:
Given that the CRT match may or may not be there, and if it's not there it should not be used, I recommend a more explicit interface. Two options:
1. a C pointer:
```suggestion
		       const sbn::crt::MatchedCRT * crtmatch,
```
that is assigned to the address of the match (`&crthitmatch`) if there is one, or to `nullptr` if there is no match.

2. a `std::optional` object (`#include <optional>`):
```suggestion
		       std::optional<sbn::crt::MatchedCRT> crtmatch,
```
which is assigned a `crthitmatch` object when available, or `std::nullopt` otherwise. A disadvantage of this one is that it copies the object, while the pointer does not. Semantically, `optional` is the most correct approach, i.e. if I read the signature of that function I _know_ that `crtmatch` may or may not be there, while with a C pointer I don't know whether it's acceptable or not for it not to be there — and with a reference, as it is now, I know it _must_ be there.

Both pointer and `optional` have a similar interface (access elements with member access operator `->`, check if it's valid or not by implicit or explicit conversion to `bool`, like in `if (crtmatch) ...`).